### ISSUE TITLE
Issue: FactoryMetaClass does not define __getitem__

### ISSUE THREAD

**Issue by Tsdevendra1** [youtux/types-factory-boy#53]
I'm not sure if I'm using this wrong but when I define a factory such as:

```

class UserFactory(factory.django.DjangoModelFactory[User]):
    class Meta:
        model = User
```

I get the following error:

`Class 'FactoryMetaClass' does not define '__getitem__', so the '[]' operator cannot be used on its instances `

I understand why the error occurs, but I'm not sure what the intended solution is. Since DjangoModelFactory is generic in the stubs, what's the intended usage if DjangoModelFactory isn't actually able to support __getitem__? Should I monkeypatch FactoryMetaClass?

**Comment by axieum** [comment]
It looks like typeddjango/django-stubs monkey patch Django itself, is this needed here?

https://github.com/typeddjango/django-stubs/blob/febc04e27a53c030d1220d6c2cd543df9b010045/ext/django_stubs_ext/patch.py#L81

**Comment by axieum** [comment]
If you're also using [typeddjango/django-stubs](https://github.com/typeddjango/django-stubs), a workaround is to piggy-back off of their monkey-patching:

```py
import django_stubs_ext
import factory
from django_stubs_ext.patch import MPGeneric

# Monkeypatching Django, so stubs will work for all generics, see: https://github.com/typeddjango/django-stubs
django_stubs_ext.patch._need_generic.extend(
  [
    MPGeneric(factory.django.DjangoModelFactory),
    MPGeneric(factory.Faker),
    MPGeneric(factory.SelfAttribute),
    MPGeneric(factory.Sequence),
    MPGeneric(factory.SubFactory),
    MPGeneric(factory.Trait),
    # ...
  ]
)
django_stubs_ext.monkeypatch()
```

**Comment by youtux** [comment]
yeah it looks like this is quite a big limitation, until the upstream (django) will fix their classes to be typed and generic
